Use sudo build and manually install clang-3.7

owen-mc-diffblue opened the pull request:
Travis have removed llvm-toolchain-precise-3.7 from their whitelist
of sources for the "apt:" option, so we have to use a sudo build and
add it as a source ourselves.

Commit:
Use sudo build and manually install clang-3.7

Travis have removed llvm-toolchain-precise-3.7 from their whitelist
of sources for the "apt:" option, so we have to use a sudo build and
add it as a source ourselves.

tautschnig commented:
I have restarted the build (two times by now), and the Doxygen job keeps falling over (the first one was with a different message, though). Is this an issue with Travis' infrastructure, or is the APT key that Google provides genuinely outdated?

owen-mc-diffblue commented:
It's possible it's something to do with this commit: https://github.com/travis-ci/apt-source-safelist/commit/9cfe4d46f3fcb803ee285905946d191ab52e85a6#diff-eb3790820e97ba4b0dd22623188e706a . I don't really understand how it works.

Commit:
Try to fix travis failure on doxygen stage

The Travis log included the following lines:

W: GPG error: http://packages.cloud.google.com/apt cloud-sdk-trusty InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6A030B21BA07F4FB
W: The repository 'http://packages.cloud.google.com/apt cloud-sdk-trusty InRelease' is not signed.
W: There is no public key available for the following key IDs:
6A030B21BA07F4FB
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
travis_fold:end:apt-get.diagnostics
[0K
[31;1mThe command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends $TRAVIS_APT_OPTS install cmake google-cloud-sdk graphviz" failed and exited with 100 during .[0m

owen-mc-diffblue commented:
@tautschnig I have added a commit which seems to have fixed that problem. I must admit that I don't understand repositories and how they relate to travis all that well, so it's very much a case of "when you have a hammer everything looks like a nail".

tautschnig commented:
To me this looks like Travis is doing some sort of caching of keys, which apparently is broken. Or maybe it's https://github.com/travis-ci/apt-source-safelist/issues/388. Either way the workaround seems ok to me, just manually doing what the APT add-on was supposed to do.
